core/src/main/java/com/linecorp/armeria/common/util/Unwrappable.java

```diff
@@ -87,4 +87,37 @@ default <T> T as(Class<T> type) {
     default Unwrappable unwrap() {
         return this;
     }
+
+    /**
+     * Unwraps this object and returns the innermost object being decorated.
+     * If this {@link Unwrappable} is the innermost object, this method returns itself. For example:
+     * <pre>{@code
+     * class Foo implements Unwrappable {}
+     *
+     * class Bar<T extends Unwrappable> extends AbstractUnwrappable<T> {
+     *     Bar(T delegate) {
+     *         super(delegate);
+     *     }
+     * }
+     *
+     * class Qux<T extends Unwrappable> extends AbstractUnwrappable<T> {
+     *     Qux(T delegate) {
+     *         super(delegate);
+     *     }
+     * }
+     *
+     * Foo foo = new Foo();
+     * assert foo.unwrapAll() == foo;
+     *
+     * Bar<Foo> bar = new Bar<>(foo);
+     * assert bar.unwrapAll() == foo;
+     *
+     * Qux<Bar<Foo>> qux = new Qux<>(bar);
+     * assert qux.unwrap() == bar;
+     * assert qux.unwrapAll() == foo;
+     * }</pre>
+     */
+    default Unwrappable unwrapAll() {
+        return unwrap();
```

Comment:
Should we improve the default implementation to unwrap until the returned value is the same as the old value?
For example:
```java
Unwrappable unwrapped = this;
while (true) {
    final Unwrappable inner = unwrapped.unwrap();
    if (inner == unwrapped) {
        return inner;
    }
    unwrapped = inner;
}
```

Reply:
Good point 👍 I think the API makes more sense this way in case an implementation overrides `Unwrappable#unwrap`.
  Also added a test case to preserve this behavior.